CMakeLists.txt

```diff
@@ -119,11 +122,17 @@ ELSE()
   ENDIF()
 ENDIF()
 
-# Force Runtime output directory into build/bin folder; force multiprocessor compilation for all projects
+# Force Runtime output directory into Debug or Release folder
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/$<CONFIG> CACHE PATH "")
 IF(MSVC)
-  set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin/$<CONFIG> CACHE PATH "")
-  set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin/$<CONFIG> CACHE PATH "")
+  # Put shared libraries in the same directory as the executable on windows
+  set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/$<CONFIG> CACHE PATH "")
+  # Force multiprocessor compilation for all projects
   add_compile_options($<$<CXX_COMPILER_ID:MSVC>:/MP>)
+ELSEIF(UNIX)
+  # Put shared libraries in ./lib, and use RPATH
+  set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/$<CONFIG>/lib CACHE PATH "")
```

Comment:
i'm skeptical about this.
Why bother on linux?

Reply:
I'm thinking it would just be cleaner to not have the directory cluttered with shared libraries.